packages/semcom-components/lib/components/sem-com-register.component.ts

```diff
@@ -0,0 +1,123 @@
+
+  render(): TemplateResult {
+
+    let componentToRender = html``;
+
+    if (this.state && this.state.matches(SemComRegisterStates.AUTHENTICATING)) {
+
+      componentToRender = html`<auth-flow .solidService="${this.solidService}" @authenticated="${this.onAuthenticated}"></auth-flow>`;
+
+    } else if (this.state && this.state.matches(SemComRegisterStates.STORE_SELECTION)) {
+
+      componentToRender = html`<sem-com-store-selection .actor="${this.actor}" .semComStoreUrls="${[ 'http://localhost:3002/testpod1/', 'http://localhost:3002/testpod2/' ]}"></sem-com-store-selection>`;
+
+    } else if (this.state && this.state.matches(SemComRegisterStates.CHECKING_PERMISSION)) {
+
+      componentToRender = html`<loading-component message="Checking permissions..."></loading-component>`;
+
+    } else if (this.state && this.state.matches(SemComRegisterStates.UPLOAD_COMPONENT_FORM)) {
+
+      componentToRender = html`<sem-com-upload-form .actor="${this.actor}"></sem-com-upload-form>`;
+
+    } else if (this.state && this.state.matches(SemComRegisterStates.NOT_PERMITTED)) {
+
+      componentToRender = html`<feedback-component id="fc-store-selection-error" title="An error occured!" message="You do not have permission to edit this store! Please choose another store or log in with a user account that has access to this store." buttonText="Go back to the store selection screen"></feedback-component>`;
+      this.addEventListenerAfterUpdate('fc-store-selection-error', () => this.actor.send(new BackToStoreSelectionEvent()));
+
+    } else if (this.state && this.state.matches(SemComRegisterStates.UPLOADING_COMPONENT)) {
+
+      componentToRender = html`<loading-component message="Saving data..."></loading-component>`;
+
+    } else if (this.state && this.state.matches(SemComRegisterStates.SUCCESSFULLY_SAVED_DATA)) {
+
+      componentToRender = html`<feedback-component id="fc-upload-successful" success title="Data saved successfully!" buttonText="Add another component"></feedback-component>`;
+      this.addEventListenerAfterUpdate('fc-upload-successful', () => this.actor.send(new BackToUploadFormEvent()));
+
+    } else if (this.state && this.state.matches(SemComRegisterStates.ERROR_SAVING_DATA)) {
+
+      componentToRender = html`<feedback-component id="fc-error-saving-data" title="An error occurred!" message="Something went wrong while trying to save the data to the desired store. Please try again." buttonText="Go back to the store selection screen"></feedback-component>`;
+      this.addEventListenerAfterUpdate('fc-error-saving-data', () => this.actor.send(new BackToStoreSelectionEvent()));
+
+    }
+
+    return html`
+        <div class="root-content">
+          <nav>
+            <a class="link-item">
+              <img class="logo" src="logo-here.png">
+              <span class="title">title</span>
+            </a>
+          </nav>
```

Comment:
This is not really needed. The surrounding layout in Figma serves as an example, but is not part of the component itself (rather it should be in the demo here)